vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/test/java/com/vaadin/flow/component/combobox/MultiSelectComboBoxTest.java

```diff
@@ -220,4 +222,43 @@ public void getSelectedItems_returnsImmutableSet() {
 
         comboBox.getSelectedItems().add("baz");
     }
+
+    @Test
+    public void preserveOrderTest() {
+        MultiSelectComboBox<String> select = new MultiSelectComboBox<>();
+        select.setItems("One", "Two", "Three", "Four", "Five", "Six", "Seven",
+                "Eight", "Nine", "Ten");
+
+        Set<String> value = null;
+        List<String> list1 = null;
+
+        select.select("Eight");
+        select.select("Two");
+        select.select("Four");
+        value = select.getValue();
+        list1 = value.stream().collect(Collectors.toList());
+        Assert.assertEquals("Eight", list1.get(0));
+        Assert.assertEquals("Two", list1.get(1));
+        Assert.assertEquals("Four", list1.get(2));
+        select.clear();
+
+        Set<String> list2 = new LinkedHashSet<>(
```

Comment:
```suggestion
        Set<String> linkedHashSetValue = new LinkedHashSet<>(
```
Let's give this a purposeful name.

Reply:
Done